client/components/widget/query/query-result-data-service.js

```diff
@@ -233,19 +233,29 @@ QueryResultDataService.prototype.fetchResults = function(widget) {
         let rows = response.data.totalRows;
         let size = response.data.totalBytesProcessed;
         let time = response.data.elapsedTime;
-        let job = response.data.jobReference.jobId;
+        let jobReference = response.data.jobReference;
 
-        widget.state().datasource.job_id = job;
+        if (goog.isDefAndNotNull(jobReference)) {
```

Comment:
Please add a TODO for the refactoring, otherwise this is fine - I will pull it in.